### ISSUE TITLE
warning problem

### ISSUE THREAD

**Issue by jokercao6** [wanghao9610/OV-DINO#57]
Has anyone encountered this warning? 
How to solve it? 
Although it seems that it does not affect the final result
------------------------------------------------------------
.local/lib/python3.10/site-packages/torch/optim/lr_scheduler.py:224: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the o
pposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/opt
im.html#how-to-adjust-learning-rate

**Comment by wanghao9610** [comment]
Hi, @jokercao6 you can ignore this warning, and the training pipeline is compatible with the popular detectron2 repo.

**Comment by wanghao9610** [comment]
Hi, @jokercao6 you can ignore this warning, and the training pipeline is compatible with the popular detectron2 repo.

**Comment by jokercao6** [comment]
> Hi, [@jokercao6](https://github.com/jokercao6) you can ignore this warning, and the training pipeline is compatible with the popular detectron2 repo.

Oh, thank you.
I also have another question. When I reproduce the results of fine-tuning the coco dataset, I basically cannot achieve the same results as the paper, usually 0.2 lower.
Then I modified the training dataset to the normal version (w/o unipro), but it was higher than the unipro I reproduced. I would like to ask, do you have any special experimental settings when fine-tuning?

![Image](https://github.com/user-attachments/assets/b27d7ae3-c5bf-45e6-a802-abd24f7dcb49)

**Comment by wanghao9610** [comment]
Hi, @jokercao6,

Thanks for your attention to our work. The performance of a single experiment may exhibit slight fluctuations, which is normal in deep learning due to environmental differences and random conditions (even if the package versions are successfully matched). I have noticed that the results of OV-DINO^1 and OV-DINO^2 are comparable, while only OV-DINO^3 shows a significant gap compared to what was reported in our paper.

You may need to check the configuration file and print the category_names to ensure the formatted category_names are as intended. From my memory (about six months ago), OV-DINO^3 without UniPro leads to a significant drop in performance, as the pre-training is with UniPro.

I will consider releasing the fine-tuning logs on COCO in the next few days.

**Comment by jokercao6** [comment]
> Hi, [@jokercao6](https://github.com/jokercao6),
> 
> Thanks for your attention to our work. The performance of a single experiment may exhibit slight fluctuations, which is normal in deep learning due to environmental differences and random conditions (even if the package versions are successfully matched). I have noticed that the results of OV-DINO^1 and OV-DINO^2 are comparable, while only OV-DINO^3 shows a significant gap compared to what was reported in our paper.
> 
> You may need to check the configuration file and print the category_names to ensure the formatted category_names are as intended. From my memory (about six months ago), OV-DINO^3 without UniPro leads to a significant drop in performance, as the pre-training is with UniPro. Additionally, the configuration files in the repository for COCO fine-tuning are designed for 8 A100 GPUs with a batch size of 32. Please ensure that these settings are matched.
> 
> I will consider releasing the fine-tuning logs on COCO in the next few days.

Thank you for your reply, these two configurations may not have such a big impact. 
Happy Chinese New Year
